Classes/Utility/GeneralUtility.php

```diff
@@ -413,19 +413,21 @@ public static function fullUrl($url, $domain = null)
             if ($url === '/') {
                 $url = '';
             }
-
-            // remove first /
-            if (strpos($url, '/') === 0) {
-                $url = substr($url, 1);
-            }
-
+            
             if ($domain !== null) {
-                // specified domain
-                $url = 'http://' . $domain . '/' . $url;
+                if(!preg_match('/^https?:\/\//i', $domain)) {
+                    $domain = 'http://' . $domain;
```

Comment:
If the user of this method really wants to specify the scheme, it could be done with the domain parameter. Maybe it's more clear to rename the parameter from $domain to something like $baseUrlOverride